crypto/merkle/simple_proof_test.go

```diff
@@ -39,3 +41,33 @@ func TestSimpleProofValidateBasic(t *testing.T) {
 		})
 	}
 }
+
+func TestSimpleProofProtoBuf(t *testing.T) {
+	testCases := []struct {
+		msg     string
+		ps1     *SimpleProof
+		expPass bool
+	}{
+		{"failure empty", &SimpleProof{}, false},
+		{"failure nil", nil, false},
+		{"success",
+			&SimpleProof{
+				Total:    1,
+				Index:    1,
+				LeafHash: tmrand.Bytes(32),
+				Aunts:    [][]byte{tmrand.Bytes(32), tmrand.Bytes(32), tmrand.Bytes(32)},
+			}, true},
+	}
+
+	for _, tc := range testCases {
```

Comment:
It's conventional for table-driven tests to have a `t.Run()` subtest for each case. Protip: `tdt<enter>` in VSCode will generate the skeleton for you.

Reply:
i need to figureout  my vscode setup, snippets dont work :/